pkg/git/gogit.go

```diff
@@ -297,6 +314,21 @@ func (g *GoGit) GetRemoteUrl(dir string, remoteName string) (string, error) {
 	return utils.SanitizeRepoUrl(urls[0]), nil
 }
 
+func (g *GoGit) ValidateAccess(ctx context.Context, url string, branch string) error {
+
+	path, err := ioutil.TempDir("", "temp-src")
+	if err != nil {
+		return fmt.Errorf("error creating temporary folder %w", err)
+	}
+	defer os.RemoveAll(path)
+
+	_, err = g.clone(ctx, path, url, branch, 1)
+	if err != nil && !errors.Is(err, transport.ErrEmptyRemoteRepository) {
```

Comment:
What about `ErrRepositoryAlreadyExists`, do we want to ignore that error as well?

Reply:
Good point. It seems we can only reproduce that error if we call Init or/and Clone twice using the same path https://github.com/go-git/go-git/blob/master/repository_test.go#L122. This call to clone always use a random folder, probably this error is not very likely to occur 🤔. But I could be missing something.

Reply:
It might be worth actually checking, it's a low friction add and the "probably this error is not very likely to occur" will get us eventually :)

Reply:
sounds good

Reply:
after further research, we found out that `PlainCloneContext` could return `ErrRepositoryAlreadyExists` before doing an actual ssh key validation so we don't want to ignore it.

Reply:
Nice work digging into this @josecordaz